sound/soc/sof/intel/hda-dai.c

```diff
@@ -466,18 +468,30 @@ static int ipc4_hda_dai_trigger(struct snd_pcm_substream *substream,
 
 	w = snd_soc_dai_get_widget(dai, substream->stream);
 	swidget = w->dobj.private;
+	pipe_widget = swidget->pipe_widget;
+	pipeline = pipe_widget->private;
 
 	switch (cmd) {
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		snd_hdac_ext_stream_start(hext_stream);
+		if (pipeline->state != SOF_IPC4_PIPE_PAUSED) {
```

Comment:
Is there a chance that the pipeline is in RUNNING state?
A second FE connects to the already running path via a mixer for example?
Will a pause/run toggle going to affect the already running stream?

Reply:
Humm, not so sure @ranj063 . What I did is prevent the triggers from happening twice at the BE dailink level. 

This is the dai level. We should double-check that indeed this is not going to happen.

Reply:
@plbossart Im 100% sure this works as expected. Please take a look at https://github.com/thesofproject/linux/blob/ea1ca7d0610c8cc4cd8101e239694828af20da7e/sound/soc/soc-pcm.c#L2118

Reply:
yes indeed. the PCM is a level higher than both dailink and dai.